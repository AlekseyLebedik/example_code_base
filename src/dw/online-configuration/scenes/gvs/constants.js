export const urlPattern = '/online-configuration/:titleId/:env';
export const SCENES = {
  DRAFTS: 'drafts',
  EVENTS: 'events',
  CONFIGURATION: 'configuration',
  VARIABLE_DEFINITIONS: 'variable-definition',
  COMPARE: 'compare',
  GROUPS: 'groups',
};
export const gvsUrlPattern = `${urlPattern}/gvs/:scene(${Object.values(
  SCENES
).join('|')})/:scopeURI?`;

export const MAX_DRAFT_NAME_LENGTH = 64;

export const PLAYER = 'player_p';

export const GLOBAL_PLAYERS = 'global:player_p';
export const GLOBAL_DEDI = 'global:dedi';
export const GLOBAL_UDP = 'global:udp';

export const GLOBAL_NON_PLAYER = [GLOBAL_DEDI, GLOBAL_UDP];

export const POPULATION_TYPE_OPTIONS = [
  { value: GLOBAL_PLAYERS, label: 'Players' },
  { value: GLOBAL_DEDI, label: 'Dedis' },
  { value: GLOBAL_UDP, label: 'UDP Relay' },
];

export const POPULATION_TYPE_LABELS = POPULATION_TYPE_OPTIONS.reduce(
  (acc, { value, label }) => ({ ...acc, [value]: label }),
  {}
);

export const DEFAULT_POPULATIONS = {
  player_p: [{ value: 'global:player_p', label: 'global' }],
  dedi: [{ value: 'global:dedi', label: 'global' }],
  udp: [{ value: 'global:udp', label: 'global' }],
};
