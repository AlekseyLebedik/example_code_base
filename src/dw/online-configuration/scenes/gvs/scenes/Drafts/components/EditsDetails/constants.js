import cn from 'classnames';
import styles from './index.module.css';

export const COLUMN_DEFS = [
  {
    field: 'createdAt',
    headerName: 'Date',
    sort: 'desc',
    valueFormatter({ value, context: { formatDateTime } }) {
      return formatDateTime(value);
    },
    minWidth: 280,
    cellRenderer: 'agGroupCellRenderer',
    cellClass({ data }) {
      if (data?.isNew === undefined) return '';
      return cn(styles.new, data.isNew ? styles.newVisible : styles.newHidden);
    },
  },
  {
    field: 'createdBy',
    headerName: 'Editor',
    valueGetter: ({ data }) => data.createdBy.username,
    width: 200,
  },
  {
    field: 'comment',
    headerName: 'Comment',
    flex: 1,
    minWidth: 250,
  },
  {
    field: 'delete',
    pinned: 'right',
    cellRenderer: 'deleteCellRenderer',
    suppressSizeToFit: true,
    width: 100,
  },
];

const cellClass = ({ value, data, colDef: { field } }) => {
  if (value === '__UNSET__') return styles.before;
  if (data?.name === 'before') {
    return field === 'name' ? styles.beforeName : styles.before;
  }
  return '';
};

const valueFormatter = ({ value }) =>
  value === '__UNSET__' ? 'Not Set' : value;

export const getDetailColumnDefs = () => [
  {
    headerName: 'Scope',
    minWidth: 180,
    showRowGroup: 'scopeName',
    cellRenderer: 'agGroupCellRenderer',
    cellRendererParams: { suppressCount: true },
  },
  {
    headerName: 'Population',
    showRowGroup: 'population',
    minWidth: 150,
    cellRenderer: 'agGroupCellRenderer',
    cellRendererParams: { suppressCount: true },
  },
  {
    field: 'name',
    minWidth: 280,
  },
  {
    field: 'platform',
  },
  {
    field: 'source',
    minWidth: 150,
    valueFormatter,
    cellClass,
  },
  {
    field: 'target',
    minWidth: 150,
    valueFormatter,
    cellClass,
  },
  { field: 'scopeURI', sort: true, hide: true },
  {
    field: 'scopeName',
    hide: true,
    rowGroup: true,
    valueGetter: ({ data }) =>
      data ? data.scopeDisplayName || data.scopeName : null,
  },
  {
    field: 'population',
    hide: true,
    rowGroup: true,
    valueGetter: ({ data }) => {
      if (!data) return undefined;
      return data.displayValue || data.population;
    },
  },
];

export const ROW_HEIGHT = 42;
export const MAX_DETAIL_LINE_LENGTH = 60;
export const DETAIL_CELL_LINE_LENGTH = 22;
export const DETAIL_CELL_LINE_HEIGHT = 18;
