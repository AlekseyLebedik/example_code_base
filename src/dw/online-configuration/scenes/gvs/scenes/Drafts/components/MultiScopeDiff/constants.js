import styles from './index.module.css';

const valueCellClass = ({ value }) =>
  value === '__UNSET__' ? styles.cellUnset : '';

export const MASTER_COLUMN_DEFS = [
  {
    field: 'scopeURI',
    headerName: 'Scope',
    sort: 'asc',
    cellRenderer: 'agGroupCellRenderer',
    valueFormatter({ value, data }) {
      return data.scopeDisplayName || value;
    },
  },
  {
    field: 'population',
    valueFormatter({ value, data }) {
      return data.displayValue || value;
    },
  },
];

const valueFormatter = ({ value }) =>
  value === '__UNSET__' ? 'Not Set' : String(value);

export const DETAIL_COLUMN_DEFS = [
  { field: 'key', minWidth: 250, maxWidth: 600, flex: 1 },
  {
    field: 'platform',
    valueFormatter({ value }) {
      return value === '*' ? 'Default' : value;
    },
    maxWidth: 150,
  },
  { field: 'source', valueFormatter, cellClass: valueCellClass },
  { field: 'target', valueFormatter, cellClass: valueCellClass },
];
