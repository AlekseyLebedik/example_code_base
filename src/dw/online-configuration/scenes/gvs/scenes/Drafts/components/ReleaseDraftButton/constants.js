export const CONFLICT_MSG = `
Since you have viewed this draft it has been updated by another user.
Please, review the complete diff for the draft again.`;

export const SUCCESS_MSG = 'Draft has been successfully released';
