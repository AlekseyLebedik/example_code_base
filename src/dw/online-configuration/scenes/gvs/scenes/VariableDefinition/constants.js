import capitalize from 'lodash/capitalize';

export const BOOL = 'bool';
export const CHAR = 'char';
export const FLOAT = 'float32';
export const INT = 'int64';

export const TYPE_OPTIONS = [
  { value: CHAR, label: 'String' },
  { value: BOOL, label: 'Boolean' },
  { value: INT, label: 'Number' },
  { value: FLOAT, label: 'Decimal' },
];

const TYPE_LABELS_MAP = TYPE_OPTIONS.reduce(
  (acc, { value, label }) => ({ ...acc, [value]: label }),
  {}
);

export const COLUMN_DEFS = [
  {
    field: 'key',
    sort: 'asc',
    checkboxSelection: true,
    headerCheckboxSelection: true,
    headerCheckboxSelectionFilteredOnly: true,
    cellRenderer: 'keyCellRenderer',
    minWidth: 350,
  },
  {
    field: 'type',
    width: 100,
    valueFormatter({ value }) {
      const plural = value.startsWith('list_');
      const formatted = value
        .split('_')
        .map(v => TYPE_LABELS_MAP[v] || v)
        .map(capitalize)
        .join(' of ');
      return plural ? `${formatted}s` : formatted;
    },
    minWidth: 150,
  },
  {
    field: 'validation',
    getQuickFilterText() {
      return null;
    },
    minWidth: 250,
  },
  {
    field: 'category',
  },
  {
    field: 'description',
    flex: 1,
  },
  {
    field: 'createdBy.username',
    displayName: 'Created By',
  },
  {
    field: 'owner.username',
    displayName: 'Owner',
  },
  {
    field: 'isArchived',
    displayName: 'Archived',
    valueFormatter({ value }) {
      return value ? 'Yes' : 'No';
    },
  },
  {
    field: 'createdAt',
    valueFormatter({ value, context: { formatDateTime } }) {
      return formatDateTime(value);
    },
    getQuickFilterText({ value, context: { formatDateTime } }) {
      return `${formatDateTime(value)} ${formatDateTime(
        value,
        'YYYY-MM-DD HH:mm:ss'
      )}`;
    },
    minWidth: 220,
  },
];
