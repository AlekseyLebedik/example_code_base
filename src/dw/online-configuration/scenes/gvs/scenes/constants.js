export const DEFAULT_PLATFORMS = [
  'atvi',
  'bnet',
  'dedr',
  'hl',
  'ps4',
  'ps5',
  'studio',
  'ul',
  'wecn',
  'xone',
  'xb1',
  'xbsx',
];
