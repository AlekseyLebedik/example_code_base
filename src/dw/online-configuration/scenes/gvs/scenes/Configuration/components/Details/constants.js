import cn from 'classnames';
import { PLAYER } from '@gvs/constants';
import styles from './index.module.css';

export const LINE_HEIGHT = 18;
export const PADDING = 16;
export const MIN_LINE_LENGTH_TO_BREAK = 60;
export const LINE_LENGTH = 22;

const valueSetter = ({
  newValue,
  context: { scopeURI, population, draftID, platforms },
  colDef: { field: platform },
  api,
  node,
}) => {
  const { valuesPerPlatform } = node.data;
  const oldValue =
    valuesPerPlatform && valuesPerPlatform[platform]
      ? valuesPerPlatform[platform].value
      : undefined;
  const isUncommitted = oldValue !== newValue;
  const [populationType, populationValue = PLAYER] = population.split(':');
  const perPlatformValue = {
    key: node.data.key,
    isUncommitted,
    value: newValue,
    platform,
    source: {
      draftID,
      populationType,
      populationValue,
      population,
      scopeURI,
    },
  };
  if (isUncommitted) {
    // eslint-disable-next-line
    node.data[platform] = perPlatformValue;
    // eslint-disable-next-line
    node.data.isUncommitted = isUncommitted;
  } else {
    // eslint-disable-next-line
    node.data[platform] =
      valuesPerPlatform && valuesPerPlatform[platform]
        ? valuesPerPlatform[platform]
        : undefined;
    // eslint-disable-next-line
    node.data.isUncommitted = platforms.some(p => node.data[p]?.isUncommitted);
  }
  api.redrawRows({ rowNodes: [node] });
  return true;
};

const valueGetter = ({ data, colDef: { field } }) => data && data[field]?.value;

const cellClass = ({
  data,
  colDef: { field },
  context: { scopeURI, population },
}) => {
  const platformValue = data && data[field];
  if (!(platformValue && platformValue.source)) return styles.base;
  const platformPopulation = [
    platformValue.source.populationType,
    platformValue.source.populationValue || PLAYER,
  ].join(':');
  const level =
    scopeURI.split(':').length -
    platformValue.source.scopeURI.split(':').length;
  let className = styles.base;
  const isDraft = Boolean(platformValue.source.draftID);
  const { isUncommitted } = platformValue;
  if (platformValue.source.populationType === 'global') {
    className = styles.global;
  } else if (population === platformPopulation) {
    className = styles.current;
  } else {
    className = styles.parent;
  }
  if (isUncommitted) className = styles.edit;
  else if (isDraft || platformValue?.source?.isUncommitted) {
    className = cn(className, styles.draft);
  }
  return cn(className, styles[`levelUp${level}`]);
};

const columnProps = {
  cellRenderer: 'platformCellRenderer',
  cellClass,
  valueSetter,
  valueGetter,
  cellEditor: 'platformCellEditor',
  minWidth: 100,
  maxWidth: 100,
};

export const getColumnDefs = (
  platforms,
  editable,
  obfuscationColumnVisible = false
) => [
  {
    field: 'key',
    headerName: 'Name',
    filter: 'agTextColumnFilter',
    pinned: 'left',
    sort: 'asc',
    cellClass({ data }) {
      if (data.isUncommitted) return styles.edit;
      if (data.draftID) return styles.draft;
      return '';
    },
    minWidth: 300,
    cellRenderer: 'nameCellRenderer',
  },
  {
    field: 'obfuscatedKey',
    headerName: 'Obfuscated Name',
    pinned: 'left',
    minWidth: 170,
    maxWidth: 170,
    initialHide: !obfuscationColumnVisible,
    valueGetter: ({ data, context }) =>
      data?.obfuscatedKey ||
      (context?.obfuscatedKeys && context.obfuscatedKeys[data?.key]),
  },
  {
    field: '*',
    headerName: 'Default Value',
    pinned: 'left',
    editable,
    ...columnProps,
    minWidth: 210,
    maxWidth: 210,
  },
  ...[...platforms]
    .sort()
    .filter(p => p !== '*')
    .map(p => ({
      field: p,
      headerName: p,
      editable,
      ...columnProps,
    })),
];
