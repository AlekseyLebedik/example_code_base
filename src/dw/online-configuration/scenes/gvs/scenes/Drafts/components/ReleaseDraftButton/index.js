import React, { useCallback, useContext, useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import {
  generatePath,
  useHistory,
  useParams,
  useRouteMatch,
} from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Icon from '@material-ui/core/Icon';
import Accordion from '@material-ui/core/Accordion';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Typography from '@material-ui/core/Typography';
import { GVS_PUBLISH_CONFIGURATION } from '@demonware/devzone-core/access/PermissionCheck/permissions';
import { useCurrentEnvPermission, useSnackbar } from 'dw/core/hooks';
import ConfirmActionComponent from 'dw/core/components/ConfirmActionComponent';

import {
  useDrafts,
  useReleaseDraft,
  useRefreshConfigQueries,
} from 'dw/online-configuration/scenes/gvs/graphql/hooks';
import { ERROR_MSG } from 'dw/online-configuration/constants';
import { gvsUrlPattern } from 'dw/online-configuration/scenes/gvs/constants';
import { CurrentDraftContext } from 'dw/online-configuration/scenes/gvs/context';
import { CONFLICT_MSG, SUCCESS_MSG } from './constants';

const useStyles = makeStyles(theme => ({
  settingsRoot: {
    marginTop: theme.spacing(3),
  },
  settingsHeader: {
    color: theme.palette.text.secondary,
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
}));

const ReleaseDraftButton = ({ OCCToken, refetchDraftDetails }) => {
  const classes = useStyles();
  const snackbar = useSnackbar();
  const { draftId } = useParams();
  const [, setCurrentDraftId] = useContext(CurrentDraftContext);
  const [comment, setComment] = useState('');
  const [sendPushMessage, setSendPushMessage] = useState(true);
  const releaseDraftMutation = useReleaseDraft();
  const { drafts } = useDrafts();
  const draftName = useMemo(() => {
    if (!drafts) return null;
    const { name } = drafts.find(d => d.id === draftId) || {};
    return name;
  }, [drafts, draftId]);
  const defaultComment = useMemo(
    () => `Releasing the draft ${draftName}`,
    [draftName]
  );
  const setRefetchQueries = useRefreshConfigQueries();

  // Redirect to the configuration after draft has been released
  const match = useRouteMatch(gvsUrlPattern);
  const history = useHistory();
  const redirect = useCallback(() => {
    history.push(
      generatePath(gvsUrlPattern, {
        ...match.params,
        scene: 'configuration',
      })
    );
  }, [match, history]);

  const onRelease = useCallback(async () => {
    try {
      const {
        data: {
          gvsReleaseDraft: { statusCode, body },
        },
      } = await releaseDraftMutation(
        draftId,
        OCCToken,
        comment || defaultComment,
        sendPushMessage
      );
      if (statusCode === 409) {
        refetchDraftDetails.current();
        snackbar.error(CONFLICT_MSG);
        return;
      }
      if (statusCode !== 201) {
        throw new Error(body);
      }
      setCurrentDraftId(null);
      setRefetchQueries();
      setTimeout(redirect, 200);
      snackbar.success(SUCCESS_MSG);
    } catch (err) {
      // eslint-disable-next-line
      console.log(String(err));
      snackbar.error(ERROR_MSG);
    }
  }, [
    snackbar,
    releaseDraftMutation,
    draftId,
    OCCToken,
    comment,
    defaultComment,
    setRefetchQueries,
    redirect,
    sendPushMessage,
    setCurrentDraftId,
  ]);
  const publishConfigurationPermission = useCurrentEnvPermission([
    GVS_PUBLISH_CONFIGURATION,
  ]);
  return publishConfigurationPermission ? (
    <ConfirmActionComponent
      onClick={onRelease}
      confirm={{
        title: 'Confirm Release',
        confirmMsg: (
          <>
            <p>Are you sure you want to Release the draft {draftName}?</p>
            <TextField
              label="Comment describing release"
              InputLabelProps={{ shrink: true }}
              placeholder={defaultComment}
              value={comment}
              onChange={e => setComment(e.target.value)}
              fullWidth
            />
            <div className={classes.settingsRoot}>
              <Accordion>
                <AccordionSummary
                  expandIcon={<Icon>expand_more</Icon>}
                  aria-controls="advanced-content"
                  id="advanced-header"
                >
                  <Typography className={classes.settingsHeader}>
                    Advanced Settings
                  </Typography>
                </AccordionSummary>
                <AccordionDetails>
                  <FormControlLabel
                    control={
                      <Checkbox
                        checked={sendPushMessage}
                        onChange={event =>
                          setSendPushMessage(event.target.checked)
                        }
                        color="primary"
                      />
                    }
                    label="Send Push Message"
                  />
                </AccordionDetails>
              </Accordion>
            </div>
          </>
        ),
        variant: 'warning',
        mainButtonLabel: 'Release',
      }}
      tooltip={
        OCCToken
          ? 'Release Draft'
          : 'You need to review draft diff before release'
      }
      component="IconButton"
      disabled={!OCCToken}
    >
      call_merge
    </ConfirmActionComponent>
  ) : null;
};

ReleaseDraftButton.propTypes = {
  OCCToken: PropTypes.string,
  refetchDraftDetails: PropTypes.object.isRequired,
};

ReleaseDraftButton.defaultProps = {
  OCCToken: null,
};

export default ReleaseDraftButton;
