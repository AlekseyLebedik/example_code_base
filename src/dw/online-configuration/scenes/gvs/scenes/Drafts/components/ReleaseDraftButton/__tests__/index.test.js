import React from 'react';

import { ERROR_MSG } from 'dw/online-configuration/constants';

import { renderWithApolloClient } from '../../../../../test-utils';
import ReleaseDraftButton from '../index';
import { CONFLICT_MSG, SUCCESS_MSG } from '../constants';

const envScopeURI = 'cod:iw:5830';

const refetchDraftDetails = { current: jest.fn() };

const mockError = jest.fn();
const mockSuccess = jest.fn();

describe('GVS ReleaseDraftButton', () => {
  const render = (rawProps = {}) => {
    refetchDraftDetails.current.mockClear();
    mockError.mockClear();
    mockSuccess.mockClear();
    const { permission, draftId, ...props } = rawProps;
    const defaultProps = {
      refetchDraftDetails,
      ...props,
    };
    return renderWithApolloClient(<ReleaseDraftButton {...defaultProps} />, {
      params: {
        env: 'dev',
        titleId: '5830',
        scopeURI: envScopeURI,
        draftId,
      },
      scopeURI: envScopeURI,
      permission,
      snackbar: { error: mockError, success: mockSuccess },
    });
  };
  describe('renders', () => {
    it('nothing if no permissions', () => {
      const wrapper = render({ permission: false });
      expect(wrapper).toMatchSnapshot();
    });
    it('disabled if no OCCToken', () => {
      const wrapper = render();
      expect(wrapper).toMatchSnapshot();
    });
    it('enabled with OCCToken', () => {
      const wrapper = render({ OCCToken: '1234567890' });
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('mutation', () => {
    it('conflict', async () => {
      const wrapper = render({ OCCToken: '1234567890', draftId: '409' });
      await wrapper.props().onClick();
      expect(refetchDraftDetails.current).toBeCalled();
      expect(mockError).toBeCalledWith(CONFLICT_MSG);
    });
    it('other error', async () => {
      const consoleSpy = jest.spyOn(console, 'log').mockImplementation();
      const wrapper = render({ OCCToken: '1234567890', draftId: '404' });
      await wrapper.props().onClick();
      expect(refetchDraftDetails.current).not.toBeCalled();
      expect(mockError).toBeCalledWith(ERROR_MSG);
      expect(consoleSpy).toBeCalledWith(String(Error('NOT_FOUND')));
    });
    it('success', async () => {
      const wrapper = render({ OCCToken: '1234567890', draftId: '201' });
      await wrapper.props().onClick();
      expect(refetchDraftDetails.current).not.toBeCalled();
      expect(mockSuccess).toBeCalledWith(SUCCESS_MSG);
    });
  });
});
