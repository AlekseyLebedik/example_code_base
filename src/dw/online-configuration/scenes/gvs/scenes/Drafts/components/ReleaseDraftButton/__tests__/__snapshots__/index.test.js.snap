// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GVS ReleaseDraftButton renders disabled if no OCCToken 1`] = `
<ConfirmActionComponent
  actionTrigger="onClick"
  changeCopyPermission={null}
  component="IconButton"
  confirm={
    Object {
      "confirmMsg": <React.Fragment>
        <p>
          Are you sure you want to Release the draft 
          ?
        </p>
        <WithStyles(ForwardRef(TextField))
          InputLabelProps={
            Object {
              "shrink": true,
            }
          }
          fullWidth={true}
          label="Comment describing release"
          onChange={[Function]}
          placeholder="Releasing the draft null"
          value=""
        />
        <div
          className="makeStyles-settingsRoot-1"
        >
          <WithStyles(ForwardRef(Accordion))>
            <WithStyles(ForwardRef(AccordionSummary))
              aria-controls="advanced-content"
              expandIcon={
                <WithStyles(ForwardRef(Icon))>
                  expand_more
                </WithStyles(ForwardRef(Icon))>
              }
              id="advanced-header"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-settingsHeader-2"
              >
                Advanced Settings
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(AccordionSummary))>
            <WithStyles(ForwardRef(AccordionDetails))>
              <WithStyles(ForwardRef(FormControlLabel))
                control={
                  <WithStyles(ForwardRef(Checkbox))
                    checked={true}
                    color="primary"
                    onChange={[Function]}
                  />
                }
                label="Send Push Message"
              />
            </WithStyles(ForwardRef(AccordionDetails))>
          </WithStyles(ForwardRef(Accordion))>
        </div>
      </React.Fragment>,
      "mainButtonLabel": "Release",
      "title": "Confirm Release",
      "variant": "warning",
    }
  }
  disabled={true}
  onClick={[Function]}
  tooltip="You need to review draft diff before release"
>
  call_merge
</ConfirmActionComponent>
`;

exports[`GVS ReleaseDraftButton renders enabled with OCCToken 1`] = `
<ConfirmActionComponent
  actionTrigger="onClick"
  changeCopyPermission={null}
  component="IconButton"
  confirm={
    Object {
      "confirmMsg": <React.Fragment>
        <p>
          Are you sure you want to Release the draft 
          ?
        </p>
        <WithStyles(ForwardRef(TextField))
          InputLabelProps={
            Object {
              "shrink": true,
            }
          }
          fullWidth={true}
          label="Comment describing release"
          onChange={[Function]}
          placeholder="Releasing the draft null"
          value=""
        />
        <div
          className="makeStyles-settingsRoot-1"
        >
          <WithStyles(ForwardRef(Accordion))>
            <WithStyles(ForwardRef(AccordionSummary))
              aria-controls="advanced-content"
              expandIcon={
                <WithStyles(ForwardRef(Icon))>
                  expand_more
                </WithStyles(ForwardRef(Icon))>
              }
              id="advanced-header"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-settingsHeader-2"
              >
                Advanced Settings
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(AccordionSummary))>
            <WithStyles(ForwardRef(AccordionDetails))>
              <WithStyles(ForwardRef(FormControlLabel))
                control={
                  <WithStyles(ForwardRef(Checkbox))
                    checked={true}
                    color="primary"
                    onChange={[Function]}
                  />
                }
                label="Send Push Message"
              />
            </WithStyles(ForwardRef(AccordionDetails))>
          </WithStyles(ForwardRef(Accordion))>
        </div>
      </React.Fragment>,
      "mainButtonLabel": "Release",
      "title": "Confirm Release",
      "variant": "warning",
    }
  }
  disabled={false}
  onClick={[Function]}
  tooltip="Release Draft"
>
  call_merge
</ConfirmActionComponent>
`;

exports[`GVS ReleaseDraftButton renders nothing if no permissions 1`] = `""`;
